client/blocks/video-editor/index.jsx

```diff
@@ -0,0 +1,230 @@
+
+		if ( guid ) {
+			this.props.updatePoster( guid, { atTime: Math.floor( currentTime ) } );
+		}
+	}
+
+	handleScriptLoadError = () => {
+		this.setState( { error: true } );
+	}
+
+	handleVideoLoaded = () => {
+		this.setState( { isLoading: false } );
+	}
+
+	handleUploadImageClick = () => {
+		isSelectingFrame = false;
+
+		this.setState( {
+			error: false,
+			pauseVideo: true
+		} );
+	}
+
+	handleUploadImage = ( file ) => {
+		if ( ! file ) {
+			return;
+		}
+
+		const { media } = this.props;
+		const guid = media && media.videopress_guid ? media.videopress_guid : null;
+
+		if ( guid ) {
+			this.props.updatePoster( guid, { file } );
+		}
+	}
+
+	getVideoEditorProps( posterUrl ) {
+		const { media } = this.props;
+		const videoProperties = { posterUrl };
+
+		if ( media && media.ID ) {
+			videoProperties.ID = media.ID;
+		}
+
+		return videoProperties;
+	}
+
+	renderError() {
+		const {
+			onCancel,
+			translate,
+		} = this.props;
+
+		return (
+			<Notice
+				status="is-error"
+				showDismiss={ true }
+				text={ translate( 'We are unable to edit this video.' ) }
+				isCompact={ false }
+				onDismissClick={ onCancel } />
+		);
+	}
+
+	render() {
+		const {
+			className,
+			media,
+			onCancel,
+			uploadProgress,
+			translate,
+		} = this.props;
+		const {
+			isLoading,
+			pauseVideo,
+			error,
+		} = this.state;
+
+		const classes = classNames(
+			'video-editor',
+			className,
+		);
+
+		return (
+			<div className={ classes }>
+			{ error && this.renderError() }
+
+				<figure>
+					<div className="video-editor__content">
+						<div className="video-editor__preview-wrapper">
+							<DetailPreviewVideo
+								className="video-editor__preview"
+								isPlaying={ ! pauseVideo }
+								item={ media }
+								onPause={ this.handlePause }
+								onScriptLoadError={ this.handleScriptLoadError }
+								onVideoLoaded={ this.handleVideoLoaded }
+							/>
+						</div>
+						{ uploadProgress && ! error && ! isSelectingFrame &&
+							<ProgressBar
+								isPulsing={ true }
+								total={ 100 }
+								value={ uploadProgress } />
+						}
+						<span className="video-editor__text">
+							{ translate( 'Select a frame to use as the thumbnail image or upload your own.' ) }
+						</span>
+						<VideoEditorButtons
+							isPosterUpdating={ uploadProgress && ! error }
+							isVideoLoading={ isLoading }
+							onCancel={ onCancel }
+							onSelectFrame={ this.handleSelectFrame }
+							onUploadImage={ this.handleUploadImage }
+							onUploadImageClick={ this.handleUploadImageClick }
+						/>
+					</div>
+				</figure>
+			</div>
+		);
+	}
+}
+
+export default connect(
+	( state ) => {
+		return {
+			posterUrl: getPosterUrl( state ),
+			shouldCloseModal: shouldCloseVideoEditorModal( state ),
+			shouldShowError: shouldShowVideoEditorError( state ),
+			uploadProgress: getPosterUploadProgress( state ),
+		};
+	},
+	dispatch => bindActionCreators( {
+		updatePoster,
+	}, dispatch ),
```

Comment:
If we're not doing anything custom, mapDispatchToProps can also be an object. This can be shortened to `{ updatePoster }`. See also https://github.com/reactjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options